### PR TITLE
Support for AllTalk V1 and V2

### PR DESCRIPTION
**Pull request by erew123**
Select the version in the interface. 
I am providing an update to the AllTalk extension so that it works with both AllTalk V1 servers and AllTalk v2 servers. 

## Checklist:

- [x ] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

File: public/scripts/extensions/tts/alltalk.js [hunk 1]
```diff
@@ -1,840 +1,1113 @@
...
+
+    //######################################################//
+    // Get Current AT Server Config & Update ST (Languages) //
+    //######################################################//
+
+    updateLanguageDropdown() {
+        const languageSelect = document.getElementById('language_options');
+        if (languageSelect) {
+            // Ensure default language is set
+            this.settings.language = this.settings.language;
```
Comment (Cohee1207) [inline public/scripts/extensions/tts/alltalk.js]: This line has no effect.

File: public/scripts/extensions/tts/alltalk.js [hunk 2]
```diff
@@ -1,840 +1,1113 @@
...
+                    updateStatus('Ready');
+                } catch (error) {
+                    console.error("POST request error:", error); // Debugging statement
+                    // Set status to Error in case of failure
+                    updateStatus('Error');
+                }
+            });
+        }
+
+        function lowvramdebounce(func, delay) {
```
Comment (Cohee1207) [inline public/scripts/extensions/tts/alltalk.js]: We have a shared debounce utility function that you can import from scripts/utils.js

**Review by Cohee1207 — CHANGES_REQUESTED** [review]
1. Please provide a cleaner diff. This is hard to review without external tools, this is required to make sure that we're not losing any bugfixes made in the code base after the initial commit.
2. Please run eslint (`npm run lint`) on the changes and fix the highlighted issues. The consistent code style is important for maintainability.

**Comment by erew123** [comment]
Hi @Cohee1207 

Hopefully, this should cover everything, but let me know if not!

Thanks

erew123

### Updates in the extension
1. Core Feature Addition:
   - Added AllTalk V1/V2 server version selection
   - Added RVC (Realistic Voice Conversion) support for V2
   - RVC features are automatically disabled when V1 is selected

2. Code Improvements:
   - Replaced custom debounce implementation with shared utils.js debounce utility
   - Fixed linting issues:
     - Converted HTML attribute quotes in template literals to single quotes
     - Added trailing commas where required
     - Fixed console.error message formatting

3. New Settings/Properties Added:
   ```javascript
   server_version: 'v2' (default)
   rvc_character_voice: 'Disabled'
   rvc_character_pitch: '0'
   rvc_narrator_voice: 'Disabled'
   rvc_narrator_pitch: '0'
   ```

4. Bug Fixes/Improvements:
   - Better error handling for RVC voice fetching
   - Improved URL handling for V1/V2 differences in API responses
   - Enhanced settings initialization and validation

5. Structural Changes:
   - Added RVC-specific UI elements and controls
   - Added version-specific logic for API endpoints
   - Improved settings synchronization between UI and backend

**NOTE** On line 70 there is an eslint bypass:

```javascript
        // HTML template literals can trigger ESLint quotes warnings when quotes are used in HTML attributes.
        // Disabling quotes rule for this one line as it's a false positive with HTML template literals.
        // eslint-disable-next-line quotes
        let html = `<div class="at-settings-separator">AllTalk V2 Settings</div>`;
```

The reason is:
1. ESLint's quotes rule wants all strings to use single quotes
2. However, this is a template literal containing HTML, where double quotes are standard for attributes
3. I tried various solutions:
   - Using single quotes: `<div class='at-settings-separator'>`
   - Using double quotes: `<div class="at-settings-separator">`
   - Even tried escaping quotes
   But ESLint just kept flagging it as an error

**Review by Cohee1207 — APPROVED** [review]
Looks good now, thanks. Will test and merge tonight.

**Comment by Cohee1207** [comment]
Working.

**Comment by erew123** [comment]
@Cohee1207 Awesome! Thanks so much :)

**Pull request merged**